test/unit/proj_errno_string_test.cpp

```diff
@@ -0,0 +1,57 @@
+/******************************************************************************
+ *
+ * Project:  PROJ
+ * Purpose:  Unit test for proj_errno_string()
+ * Author:   Kurt Schwehr <schwehr@google.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2018, Kristian Evers.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include "proj.h"
+
+#include "gtest_include.h"
+
+namespace {
+
+TEST(ProjErrnoStringTest, Basic) {
+
+    // Test the no error result
+    ASSERT_EQ(0, proj_errno_string(0));
+
+    // Test both valid and invalid PROJ error numbers
+    ASSERT_STREQ("no arguments in initialization list", proj_errno_string(-1));
+    ASSERT_STREQ("invalid projection system error (-9999)", proj_errno_string(-9999));
```

Comment:
This is the kind of place where a gmock matcher is useful.  Match part of the message and in a separate expect, match the number.  You can then even make a range based for loop with a std::initializer_list and say something like this that should be less brittle.

```
#include <limits>

// Warning: untested code...
for (const int val : {-1, -1000, -9999, std::numeric_limits<int>::min()}) {
  // Messages are of the form:
  //     invalid projection system error (-1000)
  const char *msg = proj_errno_string(val);
  EXPECT_THAT(msg, testing::ContainsRegex("invalid projection"));
  // Do this part better.  I can't just use my normal absl.
  char buf[100];
  sprintf(buf, "(%d)", val);
  EXPECT_THAT(msg, testing::ContainsRegex("invalid projection"));
}
```